### PR TITLE
Allow Cromwell to specify PAPIv1-style machine sizing in PAPIv2 [BA-6222]

### PR DESCRIPTION
**Pull request by grsterin**

### CODE REVIEW

**Comment by grsterin** [comment]
> Looks good though I'm wondering if the name of the runtime attribute should be more Googley

You mean like `google_legacy_machine_selection`?

File: supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/PipelinesApiJobCachingActorHelper.scala [hunk 1]
```diff
@@ -20,7 +20,11 @@ trait PipelinesApiJobCachingActorHelper extends StandardCachingActorHelper {
 
   lazy val pipelinesApiCallPaths: PipelinesApiJobPaths = jobPaths.asInstanceOf[PipelinesApiJobPaths]
 
-  lazy val runtimeAttributes = PipelinesApiRuntimeAttributes(validatedRuntimeAttributes, pipelinesConfiguration.runtimeConfig)
+  lazy val runtimeAttributes = PipelinesApiRuntimeAttributes(
+    validatedRuntimeAttributes,
+    pipelinesConfiguration.runtimeConfig,
+    googleLegacyMachineSelection(jobDescriptor.workflowDescriptor)
```
Comment (cjllanwarne) [inline supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/PipelinesApiJobCachingActorHelper.scala:26]: Technically this isn't a "runtime" attribute, but I think it's fine to bundle these together in this case

File: supportedBackends/google/pipelines/v2alpha1/src/test/scala/cromwell/backend/google/pipelines/v2alpha1/MachineConstraintsSpec.scala [hunk 2]
```diff
@@ -14,25 +14,36 @@ class MachineConstraintsSpec extends FlatSpec with Matchers {
...
       // Memory / cpu ratio must be < 6.5GB, increase CPU
-      (MemorySize(13.65, MemoryUnit.GB), refineMV[Positive](1), "custom-4-14080"),
+      (MemorySize(13.65, MemoryUnit.GB), refineMV[Positive](1), false, "custom-4-14080"),
       // Memory should be an int
-      (MemorySize(1520.96, MemoryUnit.MB), refineMV[Positive](1), "custom-1-1536"),
-      (MemorySize(1024.0, MemoryUnit.MB), refineMV[Positive](1), "custom-1-1024")
+      (MemorySize(1520.96, MemoryUnit.MB), refineMV[Positive](1), false, "custom-1-1536"),
+      (MemorySize(1024.0, MemoryUnit.MB), refineMV[Positive](1), false, "custom-1-1024"),
+
+      /* Same tests as above but with legacy machine type selection (cpu and memory as specified, no adjustments, except float->int)  */
```
Comment (cjllanwarne) [inline supportedBackends/google/pipelines/v2alpha1/src/test/scala/cromwell/backend/google/pipelines/v2alpha1/MachineConstraintsSpec.scala:33]:
```suggestion
      /* Same tests as above but with legacy machine type selection (cpu and memory as specified. No 'custom machine requirement' adjustments are expected this time, except float->int)  */
```

File: supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/PipelinesApiJobCachingActorHelper.scala (hunk 1 above)
Comment (grsterin) [reply supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/PipelinesApiJobCachingActorHelper.scala:26]: I thought that "PipelinesApiRuntimeAttributes" in this case means "attributes which define the runtime environment (i.e., VM) which will be used by the certain PAPI Pipeline".

**Pull request merged**